### PR TITLE
Lockdown Microsoft.ML.TensorFlow public surface

### LINKED ISSUE #2280: Lockdown Microsoft.ML.TensorFlow public surface
We ought to internalize as much of that assembly to those things we want to be public, and finalize its public API.

### PR DESCRIPTION
**Pull request by ganik**
fixes #2280

### CODE REVIEW

**Comment by ganik** [comment]
>         public sealed class Arguments : TransformInputBase

its used on public constructor so has to be public

---
In reply to: [458827501](https://github.com/dotnet/machinelearning/pull/2312#issuecomment-458827501) [](ancestors = 458827501)

---
Refers to: src/Microsoft.ML.TensorFlow/TensorflowTransform.cs:40 in d1abbb8. [](commit_id = d1abbb8dd15944fafbc18df31263a6edd02d9f0d, deletion_comment = False)

**Comment by ganik** [comment]
>         internal TFGraph Graph => Session.Graph;

it will need a special presentation method then, as it is the internals of this object are not simple to understand

---
In reply to: [458827750](https://github.com/dotnet/machinelearning/pull/2312#issuecomment-458827750) [](ancestors = 458827750)

---
Refers to: src/Microsoft.ML.TensorFlow/TensorflowTransform.cs:149 in d1abbb8. [](commit_id = d1abbb8dd15944fafbc18df31263a6edd02d9f0d, deletion_comment = False)

**Comment by ganik** [comment]
>     /// <include file='doc.xml' path='doc/members/member[@name="TensorflowTransformer"]/*' />

no, only to Trasnformer

---
In reply to: [458827454](https://github.com/dotnet/machinelearning/pull/2312#issuecomment-458827454) [](ancestors = 458827454)

---
Refers to: src/Microsoft.ML.TensorFlow/TensorflowTransform.cs:37 in d1abbb8. [](commit_id = d1abbb8dd15944fafbc18df31263a6edd02d9f0d, deletion_comment = False)

File: src/Microsoft.ML.TensorFlow/TensorflowCatalog.cs [hunk 1]
```diff
@@ -45,23 +45,23 @@ public static class TensorflowCatalog
             => new TensorFlowEstimator(CatalogUtils.GetEnvironment(catalog), outputColumnNames, inputColumnNames, tensorFlowModel);
 
         /// <summary>
-        /// Score or Retrain a tensorflow model (based on setting of the <see cref="TensorFlowTransformer.Arguments.ReTrain"/>) setting.
-        /// The model is specified in the <see cref="TensorFlowTransformer.Arguments.ModelLocation"/>.
+        /// Score or Retrain a tensorflow model (based on setting of the <see cref="TensorFlowTransformer.Options.ReTrain"/>) setting.
+        /// The model is specified in the <see cref="TensorFlowTransformer.Options.ModelLocation"/>.
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.TensorFlow/TensorflowCatalog.cs]: >Options [](start = 75, length = 7)

Please rerun locally RegenerateEntryPointCatalog and upload changes in core_ep-list.tsv to this PR.

File: test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs [hunk 2]
```diff
@@ -433,7 +433,7 @@ public void TensorFlowTransformMNISTLRTrainingTest()
 
                 var pipe = mlContext.Transforms.Categorical.OneHotEncoding("OneHotLabel", "Label")
                     .Append(mlContext.Transforms.Normalize(new NormalizingEstimator.MinMaxColumn("Features", "Placeholder")))
-                    .Append(new TensorFlowEstimator(mlContext, new TensorFlowTransformer.Arguments()
+                    .Append(new TensorFlowEstimator(mlContext, new TensorFlowTransformer.Options()
```
Comment (Ivanidzo4ka) [inline test/Microsoft.ML.Tests/ScenariosWithDirectInstantiation/TensorflowTests.cs]: >TensorFlowEstimator [](start = 32, length = 19)

If you look on other peoples PR, i think we tend to change direct calls to estimators to mlContext.Transforms.TensorFlow

File: src/Microsoft.ML.TensorFlow/TensorflowCatalog.cs (hunk 1 above)
Comment (ganik) [reply src/Microsoft.ML.TensorFlow/TensorflowCatalog.cs]: done

---
In reply to: [253190931](https://github.com/dotnet/machinelearning/pull/2312#discussion_r253190931) [](ancestors = 253190931)

**Pull request merged**